code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -300,6 +300,16 @@
 	amount_per_transfer_from_this = 10
 	volume = 120
 	center_of_mass = list("x"=16, "y"=8)
+	var/spam_flag = 0
+
+/obj/item/reagent_containers/food/drinks/shaker/attack_self(mob/user as mob)
+	if (spam_flag == 0)
+		spam_flag = 1
+		playsound(src.loc, 'sound/items/soda_shaking.ogg', 50, 1)
+		src.add_fingerprint(user)
+		spawn(10)
```

Comment:
Try to avoid using spawn() where you can. Set it using world.time and a var or something similar. 
for example:
change spam_flag to be numerical instead of true/false
on successful shake, set spam_flag to world.time + 10
and have an if(world.time > spam_flag) instead of spam_flag == 0.